ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -261,20 +284,41 @@ private void setIntoTimestampColumnAndFormat(final Map<String, Expression> prope
       }
     }
 
-    private void validateWithClause(final Set<String> withClauseVariables) {
+    private void setSerdeOptions(final Map<String, Expression> properties) {
+      final boolean singleField = analysis.getSelectExpressionAlias().stream()
+          .filter(((Predicate<String>)KsqlSchema::isImplicitColumnName).negate())
+          .count() == 1;
```

Comment:
Why not just `analysis.getSelectExpressionAlias().size() == 3` or `analysis.getSelectExpressionAlias().size() - KsqlSchema.implicitColumnIndexes().size() == 1`?

Reply:
Because unfortunately that wouldn't always work. Sometime the schema has implicit columns and sometimes now. What's important is that there is only one non-implicit column.